packages/node/vro-language-server/src/server/request/collection/WorkspaceCollection.ts

```diff
@@ -52,28 +66,163 @@ export class WorkspaceCollection {
+
+            // Encode a message to a Buffer
+            const buffer = ActionsPack.encode(message).finish()
+
+            fs.writeFileSync(outputFile, buffer)
+        })
+    }
+
+    collectLocalData(modulesPath: string): any {
+        this.logger.info(`Collecting data from .js files in: ${modulesPath}`)
+
+        const modules: HintModule[] = []
+        const actions: HintAction[] = []
+        const filesArray = this.readJsFiles(modulesPath, [])
+
+        filesArray.forEach(action => {
+            const actionLines = action.split("\n")
+            const actionName = path.parse(actionLines[0]).name
+            const actionPath = path.parse(actionLines[0]).dir
+            const parsedActions = parse(action)
+            const parameters: ActionParameters[] = []
+            let returnType = ""
+            const separator = os.platform() == "win32" ? "\\" : "/"
+            const moduleName = actionPath
+                .split(`src${separator}main${separator}resources${separator}`)[1]
+                .replace(/^[\/\\]/, "") // remove leading slash or backslash
+                .replace(/[\/\\]$/, "") // remove trailing slash or backslash
+                .replace(/[\/\\]/g, ".") // replace remaining slashes or backslashes with dots
+
+            parsedActions[0].tags.forEach(tag => {
+                if (tag.tag === "param") {
+                    const actionParams: ActionParameters = {
+                        name: tag.name,
+                        type: tag.type,
+                        description: tag.description
+                    }
+                    parameters.push(actionParams)
+                }
+                if (tag.tag === "return") {
+                    returnType = tag.type
+                }
+            })
+
+            const actionObj: HintAction = {
+                id: actionName,
+                name: actionName,
+                moduleName: moduleName,
+                returnType: returnType,
+                description: parsedActions[0].description,
+                parameters: parameters
+            }
+            actions.push(actionObj)
+
+            const isModuleInArray = modules.some(module => module.name === moduleName)
+            if (!isModuleInArray) {
+                const newModule: HintModule = {
+                    id: moduleName,
+                    name: moduleName,
+                    actions: []
+                }
+                modules.push(newModule)
+            }
+        })
+
+        modules.forEach(module => {
+            actions.forEach(action => {
+                if (action.moduleName === module.name) {
+                    module.actions.push(action)
+                }
+            })
+        })
+
+        const payload = {
+            version: 1,
+            uuid: uuidv4(),
+            metadata: {
+                timestamp: Date.now(),
+                serverName: "",
+                serverVersion: "0.0.1",
```

Comment:
Is it going to be useful to cache the server name and version as well? Or we are not using this information?